tests/unit/widget/backcompat_core.js

```diff
@@ -0,0 +1,274 @@
+( function( $ ) {
+
+// Convert the values of the keys under option classes to hashes so they may be compared via
+// deepEquals. We convert to hash because order does not matter in a hash, just as it does not
+// matter in a class string.
+function convertClassValuesToHashes( options ) {
+	var classKey, classArray, classIndex;
+
+	if ( options.classes ) {
+		for ( classKey in options.classes ) {
+			if ( typeof options.classes[ classKey ] === "string" ) {
+				classArray = options.classes[ classKey ] ?
+					options.classes[ classKey ].split( " " ) : [];
+
+				// Overwrite the string-typed class value with a hash
+				options.classes[ classKey ] = {};
+				for ( classIndex in classArray ) {
+					options.classes[ classKey ][ classArray[ classIndex ] ] = true;
+				}
+			}
+		}
+	}
+
+	return options;
+}
+
+$.widget( "test.testwidget", {
+	options: {
+		classes: {
+			"test-testwidget": "ui-button-inline ui-shadow"
+		},
+		inline: true,
+		mini: false,
+		shadow: true,
+		corners: false
+	},
+	classProp: "test-testwidget",
+	_create: function() {
+		this._addClass( this.element, "test-testwidget" );
+		this._enhance();
+	},
+	_enhance: $.noop
+} );
```

Comment:
Your polluting the global test scope here and adding a new namespace on `$` this should be in the module setup and use the mobile NS. In teardown
```js
if ( $.mobile ) {
	delete $.mobile.testwidget;
	delete $.fn.testwidget;
}
```